tests/shared/test_arguments.py

```diff
@@ -0,0 +1,76 @@
+import sys
+from argparse import ArgumentParser
+
+import pytest
+
+from pubtools._pulp.arguments import SplitAndExtend
+
+FAKE_OPTION_VALUES = ["value1", "value2", "value3"]
+
+
+@pytest.fixture
+def parser():
+    return ArgumentParser()
+
+
+def test_split_and_extend_single_delimited_instance():
+    """Test a single option instance, with all values in a delimited list."""
+    # test one instance, all delimited using different delimiters
+    for delimiter in ",.-/":
```

Comment:
You could also consider using pytest.fixture or pytest.mark.parametrize for delimiter here.

The pytest.fixture approach might not be obvious... it'd look something like:

```python
@pytest.fixture(params=[',', '.', '-', '/'])
def any_delimiter(request):
  return request.param
```

Then any test which accepts 'any_delimiter' as an argument will be invoked once for each possible value.

Reply:
This is much cleaner, great suggestion!